src/main/java/uk/gov/justice/laa/crime/cfecrime/utils/FullMeansTestOutcome.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.justice.laa.crime.cfecrime.utils;
+
+import lombok.AccessLevel;
+import lombok.extern.slf4j.Slf4j;
+import lombok.Getter;
+import uk.gov.justice.laa.crime.cfecrime.cma.response.AssessmentResult;
+import uk.gov.justice.laa.crime.cfecrime.enums.CaseType;
+import uk.gov.justice.laa.crime.cfecrime.enums.Outcome;
+@Getter
+@Slf4j
+public class FullMeansTestOutcome {
+
+    @Getter(AccessLevel.NONE)
+    private AssessmentResult result;
+    @Getter(AccessLevel.NONE)
+    private  CaseType caseType;
+
+    private Outcome fullMeansOutcome;
+
+    public FullMeansTestOutcome(AssessmentResult result, CaseType caseType){
+        this.result = result;
+        this.caseType = caseType;
+        fullMeansOutcome = Outcome.NOT_POSSIBLE;
+        if (result != null &&
+            caseType != null) {
+            fullMeansOutcome = setFullMeansTestOutCome();
+        }
+    }
+
+    private Outcome setFullMeansTestOutCome() {
+        log.debug("Get the outcome of the Full Means Test result = {} caseType = {}", result, caseType);
+        Outcome outcome = Outcome.NOT_POSSIBLE;
+
+        if (result == AssessmentResult.INEL) {
+
+            outcome = checkOutcomeINEL();
+        }
+        if (result == AssessmentResult.PASS) {
+            outcome = checkOutcomePass();
+        }
+        if (result == AssessmentResult.FAIL) {
+
+            outcome = checkOutcomeFail();
+        }
+        log.info("Outcome of the Full Means Test: {}", outcome);
+        return outcome;
+    }
+
+    private Outcome checkOutcomeFail() {
+        Outcome outcome = Outcome.ELIGIBLE_WITH_CONTRIBUTION;
+        switch (caseType) {
+            case COMMITAL:
+            case SUMMARY_ONLY:
+            case EITHER_WAY:
```

Comment:
I think this misses a subtley of caseType==EITHER_WAY. If caseType==EITHER_WAY then we also need to check if the magistrate outcome is COMMITTED_FOR_TRIAL or not, to work out whether the case is in the magistrate court or the crown court. That is on the Miro, although perhaps not clear:
<img width="709" alt="Screenshot 2023-07-04 at 17 29 09" src="https://github.com/ministryofjustice/cfe-crime/assets/307612/091a7e92-1954-45fc-8679-0b7a671a2651">

So I think you'll need to add 'magistrate outcome' as another input to this class. In CMA it's [MagCourtOutcome](https://github.com/ministryofjustice/laa-crime-means-assessment/blob/main/crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/staticdata/enums/MagCourtOutcome.java)

Reply:
Sorted out as per Stephens input